packages/gamut-styles/src/utilities/themeValue.ts

```diff
@@ -0,0 +1,28 @@
+import { Theme } from '@emotion/react';
+import { get } from 'lodash';
+
+type PathImpl<T, K extends keyof T> = K extends string | number
+  ? T[K] extends Record<string | number, any>
+    ? T[K] extends ArrayLike<any>
+      ? K | `${K}.${PathImpl<T[K], Exclude<keyof T[K], keyof any[]>>}`
+      : K | `${K}.${PathImpl<T[K], keyof T[K]>}`
+    : K
+  : never;
+
+type Path<T> = PathImpl<T, keyof T> | keyof T;
+
+type PathValue<T, P extends Path<T>> = P extends `${infer K}.${infer Rest}`
+  ? K extends keyof T
+    ? Rest extends Path<T[K]>
+      ? PathValue<T[K], Rest>
+      : never
+    : never
+  : P extends keyof T
+  ? T[P]
+  : never;
+
+export function themeValue<P extends Path<Theme>>(
```

Comment:
I'm way of having two-word values for commonly used utilities. How about `themed`?

```suggestion
export function themed<P extends Path<Theme>>(
```